packages/flutter/lib/src/material/material_app.dart

```diff
@@ -88,6 +95,15 @@ class _MaterialAppState extends State<MaterialApp> implements BindingObserver {
 
   void didChangeSize(Size size) => setState(() { _size = size; });
 
+  void didChangeLocale(ui.Locale locale) => _updateLocaleState(locale);
+
+  void _updateLocaleState(ui.Locale locale) {
+    if (config.onLocaleChanged != null) {
+      config.onLocaleChanged(locale).then(
+          (LocaleQueryData data) => setState(() { _localeData = data; }));
```

Comment:
Before calling setState, we should check whether `mounted` is true.  If it's not true, then we can just return.  That happens when this widget gets removed from the tree before the future resolves.